marlowe-playground-client/src/Wallet.purs

```diff
@@ -0,0 +1,1285 @@
+  scrollHelpPanel
+
+scrollHelpPanel :: forall m. MonadEffect m => HalogenM State Action ChildSlots Message m Unit
+scrollHelpPanel =
+  liftEffect do
+    window <- Web.window
+    document <- toDocument <$> W.document window
+    mSidePanel <- WC.item 0 =<< D.getElementsByClassName "sidebar-composer" document
+    case mSidePanel of
+      Nothing -> pure unit
+      Just sidePanel -> do
+        scrollHeight <- E.scrollHeight sidePanel
+        setScrollTop scrollHeight sidePanel
+
+render :: forall p. State -> HTML p Action
+render state =
+  div [ classes [ ClassName "full-height" ] ]
+    [ section [ classes [ panelSubHeader, aHorizontal ] ]
+        [ div [ classes [ panelSubHeaderMain, aHorizontal ] ]
+            [ div [ classes [ ClassName "wallet-title", aHorizontal, jFlexStart ] ]
+                [ div [ classes [ ClassName "demos", spaceLeft ] ]
+                    [ small_ [ text "Created Wallets:" ]
+                    ]
+                ]
+            , ul [ classes [ ClassName "wallet-list", aHorizontal ] ]
+                (map (displayWallet walletName) wallets)
+            , addWalletButton
+            , a [ class_ (ClassName "clear-all-contracts"), onClick $ const $ Just ResetAll ] [ text "Clear All" ]
+            ]
+        , div [ classes [ panelSubHeaderSide, expanded (state ^. _showRightPanel) ] ]
+            [ a [ classes [ (ClassName "drawer-icon-click") ], onClick $ const $ Just $ ShowRightPanel (state ^. (_showRightPanel <<< to not)) ]
+                [ img [ src closeDrawerIcon, class_ (ClassName "drawer-icon") ] ]
+            ]
+        ]
+    , section [ classes [ ClassName "code-panel", ClassName "full-height" ] ]
+        [ mainPanel state
+        , rightPanel state
+        ]
+    ]
+  where
+  walletName = fromMaybe "No Wallet Open" $ state ^? (_openWallet <<< _Just <<< _name)
+
+  wallets = toArrayOf (_wallets <<< traversed) state
+
+  addWalletButton = button [ class_ (ClassName "add-wallet-button"), onClick $ const $ Just CreateWallet ] [ text "+" ]
+
+displayWallet :: forall p. PubKey -> Wallet -> HTML p Action
+displayWallet openWallet wallet =
+  li
+    [ classes
+        ( if wallet ^. _name == openWallet then
+            [ active ]
+          else
+            mempty
+        )
+    ]
+    [ a [ onClick $ const $ Just $ SelectWallet wallet ] [ text $ show wallet ]
+    ]
+
+mainPanel ::
+  forall p.
+  State ->
+  HTML p Action
+mainPanel state =
+  div [ class_ (ClassName "wallet-contents") ] case state ^. _view of
+    Home -> []
+    WalletView walletName ->
+      ( [ div [ class_ (ClassName "wallet-top-section") ]
+            [ renderAssets state
+            , renderMyRoles walletName state
+            ]
+        , hr_
+        , renderContracts state
+        , hr_
+        ]
+          <> renderContract'
+      )
+  where
+  started = fromMaybe false $ preview (_walletLoadedContract <<< _Just <<< _started) state
+
+  loaded = isJust $ view _walletLoadedContract state
```

Comment:
@krisajenkins annoyingly `is` doesn't compile, even though it says `Ask if preview prism would produce a Just.`

Reply:
What about:

``` purescript
  loaded = has _walletLoadedContract state
```

Reply:
it needs to be `started` as well as `having` been loaded

Reply:
nice